### ISSUE TITLE
support anon extension

### ISSUE THREAD

**Issue by lubennikovaav** [neondatabase/neon#4970]
We already build the extension as an example of remote extension,
but currently it is not possible to create it, because it requires superuser to install the extension:

pg_catalog is always owned by the initial superuser role. So in practice, you need to be superuser to run the SECURITY LABEL command, which means that you need to be superuser to install the extension.

https://gitlab.com/dalibo/postgresql_anonymizer/-/blob/master/anon.sql#L30 
```sql
--
-- By default, masking filter functions must be trusted
-- and we only trust functions from `pg_catalog` and `anon` namespaces
-- This can be bypassed by setting `anon.restrict_to_trusted_schemas` to false
--
SECURITY LABEL FOR anon ON SCHEMA pg_catalog IS 'TRUSTED';
```

Possible solutions:
- fork/ patch the extension and soften the security requirements.
- create it from superuser, as we already do with `pg_stat_statements`

cc @hlinnaka 

@acervantes23 @raoufchebri  , FYI

**Comment by hlinnaka** [comment]
Open a PR to upstream postgresql_anonymizer, to set `superuser=true` in the control file.

**Comment by lubennikovaav** [comment]
>Open a PR to upstream postgresql_anonymizer, to set superuser=true in the control file.

Well, it makes it more consistent, but doesn't really help us)

We need to test if this works:
> create it from superuser, as we already do with pg_stat_statements
https://github.com/neondatabase/neon/blob/main/compute_tools/src/spec.rs#L609

I hope this is only `create extension` and normal work doesn't require superuser.

**Comment by hlinnaka** [comment]
> > Open a PR to upstream postgresql_anonymizer, to set superuser=true in the control file.
> 
> Well, it makes it more consistent, but doesn't really help us)

Right, it would need to be:
```
superuser=true
trusted=true
```

Not sure if that would be appropriate for `postgresql_analyzer`. If it is, that'd be the best solution.

**Comment by acervantes23** [comment]
Hi @lubennikovaav where did we end up with this?

**Comment by vadim2404** [comment]
closed as duplicate #3796

**Comment by mhornbacher** [comment]
@lubennikovaav where you able to turn off `restrict_to_trusted_schemas`. We have some domain specific masking functions we need to implement that are blocking our adoption of Neon :(
